src/components/views/CategoriesView.js

```diff
@@ -0,0 +1,218 @@
+import React, { useState } from 'react';
+import { Loader } from '@graasp/ui';
+import {
+  makeStyles,
+  Button,
+  Chip,
+  FormControl,
+  FormHelperText,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+  InputLabel,
+} from '@material-ui/core';
+import SaveIcon from '@material-ui/icons/Save';
+import { hooks } from '../../config/queryClient';
+import {
+  CATEGORY_BUTTON_HEIGHT,
+  CATEGORY_CHIP_HEIGHT,
+  CATEGORY_SELECT_WIDTH,
+} from '../../config/constants';
+import ConfirmationDialog from '../common/ConfirmationDialog';
+
+const { useCategoryTypes, useCategories } = hooks;
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    marginLeft: theme.spacing(2),
+  },
+  chip: {
+    fontSize: 'large',
+    height: CATEGORY_CHIP_HEIGHT,
+    width: 'auto',
+    marginRight: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+  title: {
+    marginTop: theme.spacing(3),
+    marginBottom: theme.spacing(1),
+  },
+  wrapper: {
+    marginTop: theme.spacing(2),
+  },
+  button: {
+    marginLeft: theme.spacing(2),
+    height: CATEGORY_BUTTON_HEIGHT,
+  },
+  input: {
+    marginTop: theme.spacing(1),
+    display: 'flex',
+  },
+  textfield: {
+    marginRight: theme.spacing(2),
+  },
+  select: {
+    width: CATEGORY_SELECT_WIDTH,
+  },
+}));
+
+const CategoriesView = () => {
+  const classes = useStyles();
+  const { data: allCategoryTypes, isLoading } = useCategoryTypes();
+  const { data: allCateogories } = useCategories();
+
+  const categoriesByTypes = allCateogories?.groupBy(
+    (category) => category?.type,
+  );
+
+  const [newCategoryType, setNewCategoryType] = useState(null);
+  const [newCategory, setNewCategory] = useState(null);
+  const [selectedCategoryType, setSelectedCategoryType] = useState(null);
+  const [open, setOpen] = useState(false);
+  const [table, setTable] = useState(false);
+  const [id, setId] = useState(false);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  // eslint-disable-next-line no-multi-assign
+  const handleDelete = () => {
```

Comment:
It's probably better to have two different functions instead of a switch.

Reply:
I use a single handleDelete because we have a delete confirmation dialog and it's easy to just pass the function as props.